test/e2e/clusterctl_upgrade.go

```diff
@@ -691,6 +722,22 @@ func setupClusterctl(ctx context.Context, clusterctlBinaryURL, clusterctlConfigP
 	return clusterctlBinaryPath, clusterctlConfigPath
 }
 
+func getInfraMachinePool(ctx context.Context, managementClusterClient client.Client, mp expv1.MachinePool, workloadClusterNamespace, workloadClusterName string) *unstructured.Unstructured {
```

Comment:
Please move this func down below calculateExpectedMachinePoolMachineCount. A bit of topology sort really helps with function order (otherwise folks have to jump up and down to read through the control flow)